akka-http/src/main/scala/akka/http/scaladsl/server/Route.scala

```diff
@@ -56,8 +56,8 @@ object Route {
    *
    * This conversion is also implicitly available whereever a `Route` is used through [[RouteResult#routeToFlow]].
    */
-  def toFlow(route: Route)(implicit system: ActorSystem): Flow[HttpRequest, HttpResponse, NotUsed] =
-    Flow[HttpRequest].mapAsync(1)(asyncHandler(route, RoutingSettings(system), ParserSettings.forServer(system)))
+  def toFlow(route: Route)(implicit system: ClassicActorSystemProvider): Flow[HttpRequest, HttpResponse, NotUsed] =
+    Flow[HttpRequest].mapAsync(1)(asyncHandler(route, RoutingSettings(system.classicSystem), ParserSettings.forServer(system.classicSystem)))
```

Comment:
I wonder if we also want `ClassicActorSystemProvider`-based implicits for `SettingsCompanion` /cc @ennru

Reply:
The `apply(ActorSystem)` should get an overload with the provider.
Only the `ActorRefFactory` is accepted implicitly, I'm not sure what we should do there.

Reply:
> I wonder if we also want `ClassicActorSystemProvider`-based implicits for `SettingsCompanion` /cc @ennru

Yes, that seems to be a commonly used API that should work seemlessly with 2.6.

Reply:
>  Only the `ActorRefFactory` is accepted implicitly, I'm not sure what we should do there.

Not sure what that is about?

Reply:
In `SettingsCompanion` the `ActorRefFactory` is the implicit. I assume we need to change it to take `ClassicActorContextProvider` implicitly instead.

Reply:
hmm that would mean you really need the actor system and the actor context would not be sufficient anymore - since the actor context never has a different configuration than the actor system that seems fine to me, though. I'll update this PR.